### PR TITLE
feature: replacing webpack with esbuild.

### PR DESCRIPTION
**Pull request by dreadera**
I will be updating this library to use `esbuild` as part of trying to modernise our libraries.
I have broken down each commit into a section I have worked on.

### Changelog
- Fast Build Pipeline with `esbuild`.
- Removed `webpack`.
- Removed `babel`.
- `esm` Module support Added.
- Updated `eslint` dependencies & rules.
- Updated Dependencies.

### CODE REVIEW

**Review by uptickmetachu — APPROVED** [review]
LGTM!

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import ObjectTable, { ObjectCell, ObjectRow, BaseEditor, TextEditor, TextDrawer } from './object-table'
+
+export default ObjectTable
+export { ObjectCell, ObjectRow, BaseEditor, TextEditor, TextDrawer }
```
Comment (mwfister) [inline src/index.js:4]: Is this change tackling [this comment](https://github.com/uptick/react-object-list/pull/125#issuecomment-851876404)? If it is, maybe add a comment linking these PRs for posterity

**Review by mwfister — APPROVED** [review]

File: src/index.js (hunk 1 above)
Comment (dreadera) [reply src/index.js:4]: Yes it does tackle it and for sure I will :)

**Comment by dreadera** [comment]
This building method also tackles the issues I was having with [this](https://github.com/uptick/react-object-list/pull/125).

**Pull request merged**